### PR TITLE
Multi-scale Fourier Feature Neural Networks.

### PR DESCRIPTION
**Pull request by smao-astro**
Implemented
- Multi-scale Fourier Feature Networks
- Spatio-temporal Multi-scale Fourier Feature Networks
- examples/Poisson_multiscale_1d.py
- examples/wave_1d.py

Reference:
- https://github.com/PredictiveIntelligenceLab/MultiscalePINNs
- https://arxiv.org/abs/2012.10047

### CODE REVIEW

File: deepxde/maps/mffn.py [hunk 1]
```diff
@@ -0,0 +1,194 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from .fnn import FNN
+from .. import config
+from ..backend import tf
+
+
+class MFFN(FNN):
```
Comment (lululxvi) [inline deepxde/maps/mffn.py:10]: Rename as MsFNN. Also change the filename.

File: deepxde/maps/mffn.py [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+    """Multi-scale Fourier Feature Networks.
+
+    References:
+        https://arxiv.org/abs/2012.10047
+        https://github.com/PredictiveIntelligenceLab/MultiscalePINNs
+
```
Comment (lululxvi) [inline deepxde/maps/mffn.py:16]: Remove the empty line

File: deepxde/maps/mffn.py [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+        self.y = self.dense(y, self.layer_size[-1], use_bias=self.use_bias)
+
+        if self._output_transform is not None:
+            self.y = self._output_transform(self.x, self.y)
+
+        self.y_ = tf.placeholder(config.real(tf), [None, self.layer_size[-1]])
+        self.built = True
+
+
+class ST_MFFN(MFFN):
```
Comment (lululxvi) [inline deepxde/maps/mffn.py:127]: Rename to STMsFFN

File: examples/Poisson_multiscale_1d.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        train_distribution="pseudo",
+        solution=func,
+        num_test=10000,
+    )
+
+    layer_size = [1] + [100] * 3 + [1]
+    activation = "tanh"
+    initializer = "Glorot uniform"
+    net = dde.maps.MFFN(layer_size, activation, initializer, sigmas=[1, 10])
+    net.apply_feature_transform(lambda x: (x - 0.5) * 2 * np.sqrt(3))
```
Comment (lululxvi) [inline examples/Poisson_multiscale_1d.py:49]: I am curious if this is necessary

File: deepxde/maps/mffn.py [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+    """Spatio-temporal Multi-scale Fourier Feature Networks.
+
+    References:
+        https://arxiv.org/abs/2012.10047
+        https://github.com/PredictiveIntelligenceLab/MultiscalePINNs
+
```
Comment (lululxvi) [inline deepxde/maps/mffn.py:133]: Remove the empty line

File: examples/wave_1d.py [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+        num_test=10000,
+    )
+
+    layer_size = [2] + [100] * 3 + [1]
+    activation = "tanh"
+    initializer = "Glorot uniform"
+    net = dde.maps.ST_MFFN(
+        layer_size, activation, initializer, sigmas_x=[1], sigmas_t=[1, 10]
+    )
+    net.apply_feature_transform(lambda x: (x - 0.5) * 2 * np.sqrt(3))
```
Comment (lululxvi) [inline examples/wave_1d.py:76]: Again, is this necessary?

File: examples/wave_1d.py [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+import deepxde as dde
+
+A = 2
+C = 10
+
+
+def get_initial_loss(model):
+    model.compile("adam", lr=0.001, metrics=["l2 relative error"])
+    model.train(1)
```
Comment (lululxvi) [inline examples/wave_1d.py:21]: Can I train 0?

File: examples/wave_1d.py [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+    return model.sess.run(
```
Comment (lululxvi) [inline examples/wave_1d.py:22]: You can use
```python
losshistory, train_state = model.train()
```
to get the loss, instead of do `model.sess.run`

File: deepxde/maps/mffn.py (hunk 1 above)
Comment (smao-astro) [reply deepxde/maps/mffn.py:10]: MsFNN or MsFFN?

Comment (lululxvi) [reply deepxde/maps/mffn.py:10]: MsFFN

File: examples/Poisson_multiscale_1d.py (hunk 4 above)
Comment (smao-astro) [reply examples/Poisson_multiscale_1d.py:49]: The improvement is not significant in the case, I will remove it.

File: examples/wave_1d.py (hunk 6 above)
Comment (smao-astro) [reply examples/wave_1d.py:76]: It does improve (`test metric: [8.21e-01]` to `test metric: [3.55e-02]`), I suggest keep it here.

File: examples/wave_1d.py [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+import deepxde as dde
+
+A = 2
+C = 10
+
+
+def get_initial_loss(model):
+    model.compile("adam", lr=0.001, metrics=["l2 relative error"])
+    model.train(0)
+    return model.train_state.loss_train
```
Comment (lululxvi) [inline examples/wave_1d.py:22]: Use
```python
losshistory, train_state = model.train(0)
return losshistory.loss_train[0]
```

File: deepxde/maps/msffn.py [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+
+class MsFFN(FNN):
+    """Multi-scale Fourier Feature Networks.
+
+    Args:
+        sigmas: List of standard deviation of the distribution of fourier feature embeddings.
+    References:
+        https://arxiv.org/abs/2012.10047
+        https://github.com/PredictiveIntelligenceLab/MultiscalePINNs
+    """
```
Comment (lululxvi) [inline deepxde/maps/msffn.py:18]: """Multi-scale Fourier Feature Networks.
References:
...

Args:
...
